core/src/main/scala/org/scalarelational/extra/HikariSupport.scala

```diff
@@ -8,27 +8,26 @@ import org.scalarelational.model.SQLDatastore
  */
 trait HikariSupport extends SQLDatastore {
   // Automatically converts DataSources to be wrapped by HikariDataSource
-  dataSourceProperty.change.on {
-    case evt => evt.newValue match {
-      case null => // Ignore null DataSource
-      case ds: HikariDataSource => // Ignore HikariDataSource
-      case ds => {
-        val config = new HikariConfig()
-        config.setDataSource(ds)
-        val hikari = new HikariDataSource(config)
-        hikari.suspendPool()
-        dataSourceProperty := hikari
-      }
-    }
+  dataSourceProperty.filterCycles.attach {
+    case Some(ds) =>
+      val config = new HikariConfig()
+      config.setDataSource(ds)
+
+      val hikari = new HikariDataSource(config)
+      // hikari.suspendPool()  TODO Throws an exception
+
+      dataSourceProperty := hikari
+
+    case _ =>
   }
 
-  // Clean up data source after shutdown
-  override def dispose() = {
+  /** Clean up data source after shutdown */
+  override def dispose() {
     super.dispose()
 
     dataSource match {
-      case ds: HikariDataSource => ds.shutdown()
-      case _ => // Not a HikariDataSource
+      case Some(ds: HikariDataSource) => ds.shutdown()
```

Comment:
Exactly. For example, this will return `Int`:

```scala
val y = Option[Any](42)
y match {
  case Some(x: String) => "String"
  case Some(x: Int)    => "Int"
  case _               => "Other"
}
```